salt/modules/yumpkg.py

```diff
@@ -136,18 +137,37 @@ def _get_hold(line, pattern=__HOLD_PATTERN, full=True):
 
 def _yum():
     '''
-    return yum or dnf depending on version
+    Determine package manager name (yum or dnf),
+    depending on the system version.
     '''
     contextkey = 'yum_bin'
     if contextkey not in __context__:
-        if 'fedora' in __grains__['os'].lower() \
-                and int(__grains__['osrelease']) >= 22:
+        if ('fedora' in __grains__['os'].lower()
+           and int(__grains__['osrelease']) >= 22):
             __context__[contextkey] = 'dnf'
         else:
             __context__[contextkey] = 'yum'
     return __context__[contextkey]
 
 
+def _call_yum(args, **kwargs):
+    '''
+    Call yum/dnf.
+    '''
+    params = {'output_loglevel': 'trace',
+              'ignore_retcode': True,
```

Comment:
Same as with apt, I don't think we should make this the default. Turning this on means we won't get any logging when a command exits with a nonzero exit status. There are instances where this is useful, as some commands return nonzero exit status in non-error cases, but it's an exception, not the norm.

Reply:
Zypper actually doesn't do this. But I just moved it as a default behaviour (most of the time) in `yumpkg` module. By not ignoring the return code, could it be that the Salt won't succeed anymore running Yum? :laughing: